src/GithubPlugin.fs

```diff
@@ -127,27 +136,42 @@ type GithubPlugin() =
         | _ ->
             [ defaultResult ]
 
-    let tryRunApiSearch =
-           Async.Catch
-        >> Async.RunSynchronously
-        >> function
-            | Choice1Of2 result -> presentApiSearchResult result
-            | Choice2Of2 exn -> presentApiSearchExn exn
+    let tryRunApiSearch (result: Async<ApiSearchResult>) =
+        async {
+            let! res = result |> Async.Catch
 
+            return
+                match res with
+                | Choice1Of2 result -> presentApiSearchResult result
+                | Choice2Of2 exn -> presentApiSearchExn exn
+        }
```

Comment:
This can be simplified to

```fsharp
let tryRunApiSearch =
  Async.Catch
  >> Async.map (function
        | Choice1Of2 result -> presentApiSearchResult result
        | Choice2Of2 exn -> presentApiSearchExn exn)
```

Reply:
hmm why rider suggests that we don't have `Async.map` as a function?

Reply:
Ah, my bad, I thought `Async.map` was included in core...

In that case you can simplify it down to something like:

```fsharp
let tryRunApiSearch fSearch =
    async {
        match! Async.Catch fSearch with
        | Choice1Of2 result -> return presentApiSearchResult result
        | Choice2Of2 exn -> return presentApiSearchExn exn
    }
```